erxupdate/ntopnghosts.sh

```diff
@@ -0,0 +1,429 @@
+# so that they don't end up in the hostnames
+sed -i 's/\$$//' /tmp/ntopng_dhcpdiff
+#cat /tmp/ntopng_dhcpdiff
+if [[ "$DEBUG" != "" ]]; then 
+	trap - DEBUG
+	trap
+fi
+while IFS= read -r line; do
+    LASTHOSTNAME=$HOSTNAME
+	LASTMAC=$MAC
+	#echo $line
+	# we will process this file line by line to check each HOSTNAME
+    # echo "Text read from file: $line"
+    IP=`echo "$line" | cut -d ' ' -f 1`
+    HOSTNAME=`echo "$line" | cut -d ' ' -f 6`
+	if [[ "$HOSTNAME" == "?" || "$HOSTNAME" == "" ]]; then
+	    HOSTNAME="unknown"
+	fi
+	# Note that while NTOPNG doesn't require the MAC address we are including it in the hostname as hostname(macaddress) as this helps with uniqueness (iPhones and iPads
+	# for example all default to iPhone or iPad, unlike Android that uses a unique name).  We also use hostname and mac address for device name substitutions and for 
+	# reporting to telegram if you have enabled that functionality.
+	MAC=`echo "$line" | cut -d ' ' -f 2`
+    echo "Checking IP=$IP HOSTNAME=$HOSTNAME MAC=$MAC"
+	# echo "line=$line"
+	if [[ "$HOSTNAME $MAC" != "$LASTHOSTNAME $LASTMAC" ]]; then
+    	NTOPNGHOSTNAME="$HOSTNAME($MAC)"
+        # check if we need to replace the hostname with something more meaningful that we have on file
+		if [[ -f "/tmp/hostnamesubstitute" ]]; then
+            SUB=`grep -i "^$HOSTNAME $MAC" /tmp/hostnamesubstitute | cut -d ' ' -f 3-100`
+            if [[ "$SUB" != "" ]]; then
+                echo "Replacing $HOSTNAME with $SUB"
+                NTOPNGHOSTNAME="$SUB"
+            fi
+        fi
+        # submit the host to NTOPNG
+        RESULT=`curl -s -u $NTOPNGUSER:$NTOPNGPASS -H "Content-Type: application/json" -d '{"host": "'$IP'", "custom_name": "'$NTOPNGHOSTNAME'"}' "https://ntopng.iam-afghanistan.org/lua/rest/v1/set/host/alias.lua"`
+        #echo "RESULT=$RESULT"
+        # eg. {"rc_str_hr":"Success","rc_str":"OK","rsp":[],"rc":0}
+        # check the response for the submission
+        RC_STR=`echo $RESULT |jq -r '."rc_str"'`
+        #echo "RC_STR=$RC_STR"
+		echo "$HOSTNAME has a new IP address.  Result of uploading to NTOPNG result=$RC_STR"
+        if [[ "$RC_STR" != "OK" ]]; then
+            # Failed to update:  We're going to retry again now and then we'll give up.  This means that if we fail again that host will not update unless it 
+			# changes IP address as we'll never retry again.  The only thing that could cause a failure is if we have no internet connection when we run this 
+			# or if the ntopng server is down.  Note that following a reboot of the router we WILL try again to register all known hosts from the hosts file 
+			# to the ntopng server
+            RESULT=`curl -s -u $NTOPNGUSER:$NTOPNGPASS -H "Content-Type: application/json" -d '{"host": "'$IP'", "custom_name": "'$NTOPNGHOSTNAME'"}' "https://ntopng.iam-afghanistan.org/lua/rest/v1/set/host/alias.lua"`
```

Comment:
I believe this url should be parametric.

Reply:
See comment above.